### PR TITLE
[AIRFLOW-5062]  Allow ACL Header in S3Hook

### PR DESCRIPTION
**Pull request by retornam**
Allow passing in the ACL header in the AWS S3 Hook

Signed-off-by: Raymond Etornam <retornam@users.noreply.github.com>

---
Issue link: [AIRFLOW-5062](https://issues.apache.org/jira/browse/AIRFLOW-5062)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 1]
```diff
@@ -687,3 +701,17 @@ def download_file(
             s3_obj.download_fileobj(local_tmp_file)
 
         return local_tmp_file.name
+
+    def get_object_permission(self,
+                              bucket,
+                              key):
+        """
+        :param bucket: Name of the bucker in which the object is
```
Comment (feluelle) [inline airflow/providers/amazon/aws/hooks/s3.py:709]:
```suggestion
        :param bucket: Name of the bucket in which the object is
```

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 2]
```diff
@@ -514,9 +520,11 @@ def load_bytes(self,
         :param encrypt: If True, the file will be encrypted on the server-side
             by S3 and will be stored in an encrypted form while at rest in S3.
         :type encrypt: bool
+        :param acl: The acess control list for the uploaded file
```
Comment (feluelle) [inline airflow/providers/amazon/aws/hooks/s3.py:523]:
```suggestion
        :param acl: The access control list for the uploaded file
```

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 3]
```diff
@@ -687,3 +701,17 @@ def download_file(
             s3_obj.download_fileobj(local_tmp_file)
 
         return local_tmp_file.name
+
+    def get_object_permission(self,
+                              bucket,
+                              key):
+        """
```
Comment (feluelle) [inline airflow/providers/amazon/aws/hooks/s3.py:708]: Please add a title what the function does.

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 4]
```diff
@@ -687,3 +701,17 @@ def download_file(
...
+        :param bucket: Name of the bucker in which the object is
+        :type bucket: str
+        :param key: The key to get the object grantee permission.
+        :type key: str
+        """
```
Comment (feluelle) [inline airflow/providers/amazon/aws/hooks/s3.py:713]: Please add `:returns:` and `:rtype:` here. Thanks :)

**Review by feluelle — CHANGES_REQUESTED** [review]
Implementation looks good to me in general, but please update the documentation, thanks :) 
(see comments below)

**Comment by feluelle** [comment]
@retornam are you still working on this? Because there is also another PR #7726 which adds the same feature.

**Comment by retornam** [comment]
Yes. I’ll update the PR later today.

**Comment by feluelle** [comment]
Duplicate. Original is https://github.com/apache/airflow/pull/7733

**Pull request closed without merging**